crates/chain/src/persist.rs

```diff
@@ -1,33 +1,62 @@
-use core::convert::Infallible;
-
 use crate::Append;
+use alloc::boxed::Box;
+use core::fmt;
+
+#[derive(Debug)]
+/// Errors returned by [`PersistBackend`] when writing or loading changes.
+pub enum PersistBackendError {
+    /// Happens when there is an standard IO error writing to the backend file.
+    IoError,
+    /// Happens when there is an error iterating over the persistent backend file.
+    IterError,
```

Comment:
This error variant doesn't make sense to me since it only applies to backends that iter (also it's not clear what the caller is meant to do with this error).

Reply:
This error was intended to capture the error in `file_store/store.rs`. As far as I understand, not all users of `bdk_wallet` will use `file_store`, so I wasn't sure how to wrap it in the `PersistBackendError`. In `store`, this error occurs when applying the change set, so I guess `ApplyChangeError` could specify there is something wrong when attempting that action?